pkg/yamlmeta/map_key_overrides_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package yamlmeta_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/k14s/ytt/pkg/yamlmeta"
+)
+
+func TestMapKeyOverridePlainYAML(t *testing.T) {
+	t.Run("when no maps have duplicate keys, is a no op", func(t *testing.T) {
+		docSet := &yamlmeta.DocumentSet{
+			Items: []*yamlmeta.Document{{
+				Value: &yamlmeta.Map{
+					Items: []*yamlmeta.MapItem{
+						{Key: "foo", Value: 1},
+						{Key: "bar", Value: 2},
+					},
+				},
+			}},
+		}
+		expectedDocSet := docSet.DeepCopyAsNode()
+		docSet.OverrideMapKeys()
+
+		printer := yamlmeta.NewPrinterWithOpts(os.Stdout, yamlmeta.PrinterOpts{ExcludeRefs: true})
```

Comment:
nit: it's a little weird to have os.Stdout. lets throw in `nil` instead.

Reply:
Copy-pasta from `parser_test.go`. I'll fix more broadly in a separate commit. 👍🏻